opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -124,48 +125,133 @@ message MetricDescriptor {
-    // floating-point numbers.
+    // TODO: Decide if knowing that the aggregated measurements are raw
+    // measurements is important or not. Equivalent of "Snapshot" in #162.
+
+    // TODO: Decide if knowing that the Sum is monotonic is important or not,
+    // may be derived from the buckets definition.
+  }
+
+  // Represents the type of a metric that is calculated by computing a summary
+  // of all reported measurements over a time interval.
+  //
+  // A Metric of this Type MUST store its values as SummaryDataPoint.
+  message Summary {
+    // It describes the value type of the measurement used to build this
+    // aggregation.
     //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    MONOTONIC_DOUBLE = 4;
-
-    // Histogram measurement.
-    // Corresponding values are stored in HistogramDataPoint.
-    HISTOGRAM = 5;
-
-    // Summary value. Some frameworks implemented Histograms as a summary of observations
-    // (usually things like request durations and response sizes). While it
-    // also provides a total count of observations and a sum of all observed
-    // values, it calculates configurable quantiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 6;
+    // Determines the value type of the exemplars.
+    MeasurementValueType measurement_value_type = 1;
+
+    // Temporality is the temporal quality values of a metric have.
+    Temporality temporality = 2;
+
+    // TODO: Decide if knowing that the aggregated measurements are raw
+    // measurements is important or not. Equivalent of "Snapshot" in #162.
+
+    // TODO: Decide if knowing that the Sum is monotonic is important or not,
+    // may be derived from the buckets definition.
   }
 
-  // type is the type of values this metric has.
-  Type type = 4;
+  // Type is the type of the metric and determines what are the type of the
+  // reported value, as well as the relatationship to the time interval over
+  // which they are reported.
+  //
+  // TODO: Update table after the decision on:
+  // https://github.com/open-telemetry/opentelemetry-specification/issues/731.
+  // Be default metrics recording using the OpenTelemetry API are exported as
+  // (the table does not include MeasurementValueType to avoid extra rows):
+  //
+  //   Instrument         Type
+  //   ----------------------------------------------
+  //   Counter            Sum(temporality=delta;is_monotonic=true)
+  //   UpDownCounter      Sum(temporality=delta;is_monotonic=false)
+  //   ValueRecorder      Summary(temporality=delta)
+  //   SumObserver        Sum(temporality=cumulative;is_monotonic=true)
+  //   UpDownSumObserver  Sum(temporality=cumulative;is_monotonic=false)
+  //   ValueObserver      Gauge()
+  oneof type {
+    // TODO: Determine if encoding all possible values in a uint64 bitset
+    // improves performance significantly and propose that if that is the case.
```

Comment:
Good. We have pretty strong evidence that this is problematic, but it will vary by workload.

Reply:
Indeed, left a TODO to evaluate after we finish all other issues and see how much we can benefit.